src/libmongoc/tests/mock_server/mock-server.c

```diff
@@ -1016,13 +1017,23 @@ mock_server_receives_legacy_hello (mock_server_t *server,
 
    if (strcasecmp (request->command_name, "hello") &&
        strcasecmp (request->command_name, HANDSHAKE_CMD_LEGACY_HELLO)) {
+fprintf(stderr, "JFW: mock_server_receives_legacy_hello(): strings did NOT match; command name \"%s\"\n", request->command_name), fflush(stderr);
       request_destroy (request);
       return NULL;
    }
 
+/* JFW: */
+fprintf(stderr, "JFW: mongoc_test_settings_legacy_hello_expect_ApiVersion = %d\n", mongoc_test_settings_legacy_hello_expect_ApiVersion);
+if(mongoc_test_settings_legacy_hello_expect_ApiVersion) {
```

Comment:
In my tests, this is /never/ true. So I'm slightly confused as to how it ever works, TBH, and seem to remember this being needed-- but that was before I left for vacation, so now that I'm back it's possible that the phase of the moon has changed and it's always been fine. (Or perhaps I was linking against old code?)